### PR TITLE
Update release notes draft to version v1.31.0-beta.0

### PR DESCRIPTION
**Pull request by cloudmelon**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
This PR updates the Release Notes Draft to k/k v1.31.0-beta.0

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:
This is an automated PR generated from `krel The Kubernetes Release Toolbox`

### CODE REVIEW

File: releases/release-1.31/release-notes/maps/pr-122025-map.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+pr: 122025
+releasenote:
+  text: lastSuccessfullTime in cronjobs will now be set reliably
```
Comment (npolshakova) [inline releases/release-1.31/release-notes/maps/pr-122025-map.yaml:3]:
```suggestion
  text: `lastSuccessfullTime` in cronjobs will now be set reliably.
```

File: releases/release-1.31/release-notes/maps/pr-124320-map.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+pr: 124320
```
Comment (npolshakova) [inline releases/release-1.31/release-notes/maps/pr-124320-map.yaml:1]: Let's delete this map file- it's a known krel bug: https://github.com/kubernetes/release/issues/3516

File: releases/release-1.31/release-notes/maps/pr-124428-map.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+      I0421 07:33:35.110085 3814322 dns.go:149] Setting up hostNetwork to: true
+      I0421 07:33:35.110128 3814322 builder.go:121] Running '/usr/bin/kubectl --server=https://127.0.0.1:43529 --kubeconfig=/home/syash/.kube/kind-test-config --context=kind-kind --namespace=dns-319 exec dns-querier-1x
+    5hqm -- /bin/sh -x -c hostname'
+      I0421 07:33:35.222393 3814322 builder.go:146] stderr: "+ hostname\n"
+      I0421 07:33:35.222448 3814322 builder.go:147] stdout: "dns-querier-1x5hqm\n"
+      I0421 07:33:35.222472 3814322 dns.go:157] The hostname and pod metadata are same: dns-querier-1x5hqm, got: dns-querier-1x5hqm
+
+      I0421 07:33:35.222564 3814322 helper.go:121] Waiting up to 7m0s for all (but 0) nodes to be ready
+      STEP: Destroying namespace "dns-319" for this suite. @ 04/21/24 07:33:35.225
+    • [2.247 seconds]
```
Comment (npolshakova) [inline releases/release-1.31/release-notes/maps/pr-124428-map.yaml:31]: This got incorrectly parsed from https://github.com/kubernetes/kubernetes/pull/124428 

Let's replace it with:
```suggestion
    Added a testcase to check hostname and hostNetwork.
```

File: releases/release-1.31/release-notes/maps/pr-124994-map.yaml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+pr: 124994
+releasenote:
+  text: Added namespace autocompletion for kubectl config set-context command
```
Comment (npolshakova) [inline releases/release-1.31/release-notes/maps/pr-124994-map.yaml:3]:
```suggestion
  text: Added namespace autocompletion for `kubectl config set-context` command.
```

File: releases/release-1.31/release-notes/maps/pr-123587-map.yaml [hunk 5]
```diff
@@ -1,3 +1,4 @@
 pr: 123587
 releasenote:
   text: Fixed "-kube-test-repo-list" e2e flag may not take effect.
```
Comment (shecodesmagic) [inline releases/release-1.31/release-notes/maps/pr-123587-map.yaml:3]: is the first "-" in "-kube-test-repo-list" a typo?

File: releases/release-1.31/release-notes/maps/pr-123638-map.yaml [hunk 6]
```diff
@@ -1,4 +1,5 @@
 pr: 123638
 releasenote:
-  text: Graduated MatchLabelKeys/MismatchLabelKeys feature in PodAffinity/PodAntiAffinity to Beta.
+  text: Graduated MatchLabelKeys/MismatchLabelKeys feature in PodAffinity/PodAntiAffinity
```
Comment (shecodesmagic) [inline releases/release-1.31/release-notes/maps/pr-123638-map.yaml:3]:
```suggestion
  text: Graduated MatchLabelKeys/MismatchLabelKeys feature in PodAffinity/PodAntiAffinity.
```

File: releases/release-1.31/release-notes/maps/pr-125333-map.yaml [hunk 7]
```diff
@@ -0,0 +1,5 @@
+pr: 125333
+releasenote:
+  text: --custom flag in kubectl debug will be enabled by default and yaml support
+    is added
```
Comment (npolshakova) [inline releases/release-1.31/release-notes/maps/pr-125333-map.yaml:4]:
```suggestion
  text: "`--custom` flag in `kubectl debug` will be enabled by default and yaml support
    is added."
```

File: releases/release-1.31/release-notes/maps/pr-125442-map.yaml [hunk 8]
```diff
@@ -0,0 +1,4 @@
+pr: 125442
+releasenote:
+  text: Graduate JobPodFailurePolicy to GA and lock
```
Comment (npolshakova) [inline releases/release-1.31/release-notes/maps/pr-125442-map.yaml:3]: This PR also had `LockToDefault=true` set so maybe we can clarify:
```suggestion
  text: Graduated `JobPodFailurePolicy` to GA and locked it to it's default.
```

File: releases/release-1.31/release-notes/maps/pr-124994-map.yaml (hunk 4 above)
Comment (shecodesmagic) [inline releases/release-1.31/release-notes/maps/pr-124994-map.yaml:3]:
```suggestion
  text: Added namespace autocompletion for kubectl config set-context command.
```

File: releases/release-1.31/release-notes/maps/pr-125751-map.yaml [hunk 9]
```diff
@@ -0,0 +1,4 @@
+pr: 125751
+releasenote:
+  text: ElasticIndexedJob is graduated to GA
```
Comment (npolshakova) [inline releases/release-1.31/release-notes/maps/pr-125751-map.yaml:3]:
```suggestion
  text: "`ElasticIndexedJob` is graduated to GA."
```

File: releases/release-1.31/release-notes/maps/pr-125328-map.yaml [hunk 10]
```diff
@@ -0,0 +1,4 @@
+pr: 125328
+releasenote:
+  text: Add a warning log, an event for cgroup v1 usage and a metric for cgroup version.
```
Comment (shecodesmagic) [inline releases/release-1.31/release-notes/maps/pr-125328-map.yaml:3]:
```suggestion
  text: Added a warning log, an event for cgroup v1 usage and a metric for cgroup version.
```

File: releases/release-1.31/release-notes/maps/pr-125894-map.yaml [hunk 11]
```diff
@@ -0,0 +1,4 @@
+pr: 125894
+releasenote:
+  text: Kubernetes is now built with go 1.22.5
```
Comment (npolshakova) [inline releases/release-1.31/release-notes/maps/pr-125894-map.yaml:3]:
```suggestion
  text: Kubernetes is now built with go 1.22.5.
```

File: releases/release-1.31/release-notes/maps/pr-125970-map.yaml [hunk 12]
```diff
@@ -0,0 +1,6 @@
+pr: 125970
+releasenote:
+  text: Add `/sys/devices/virtual/powercap` to default masked paths. It avoids the
+    potential security risk that the ability to read these files may offer a power-based
+    sidechannel attack against any workloads running on the same kernel.
```
Comment (npolshakova) [inline releases/release-1.31/release-notes/maps/pr-125970-map.yaml:5]:
```suggestion
  text: Added `/sys/devices/virtual/powercap` to default masked paths. It avoids the
    potential security risk that the ability to read these files may offer a power-based
    sidechannel attack against any workloads running on the same kernel.
```

File: releases/release-1.31/release-notes/maps/pr-126012-map.yaml [hunk 13]
```diff
@@ -0,0 +1,4 @@
+pr: 126012
+releasenote:
+  text: Stop using wmic on Windows to get uuid in the kubelet
```
Comment (npolshakova) [inline releases/release-1.31/release-notes/maps/pr-126012-map.yaml:3]:
```suggestion
  text: Stoped using wmic on Windows to get uuid in the kubelet.
```

File: releases/release-1.31/release-notes/maps/pr-126046-map.yaml [hunk 14]
```diff
@@ -0,0 +1,4 @@
+pr: 126046
+releasenote:
+  text: Use omitempty for optional Job Pod Failure Policy fields
```
Comment (npolshakova) [inline releases/release-1.31/release-notes/maps/pr-126046-map.yaml:3]:
```suggestion
  text: Use omitempty for optional Job Pod Failure Policy fields.
```

File: releases/release-1.31/release-notes/maps/pr-126057-map.yaml [hunk 15]
```diff
@@ -0,0 +1,5 @@
+pr: 126057
+releasenote:
+  text: Fixes a 1.30.0 regression in openapi descriptions of PodIP.IP  and HostIP.IP
```
Comment (shecodesmagic) [inline releases/release-1.31/release-notes/maps/pr-126057-map.yaml:3]:
```suggestion
  text: Fixed a 1.30.0 regression in OpenAPI descriptions of `PodIP.IP` and `HostIP.IP`.
```

File: releases/release-1.31/release-notes/maps/pr-126057-map.yaml [hunk 16]
```diff
@@ -0,0 +1,5 @@
...
+    fields to mark the fields used as keys in those lists as required.
```
Comment (npolshakova) [inline releases/release-1.31/release-notes/maps/pr-126057-map.yaml:4]:
```suggestion
  text: Fixed a 1.30.0 regression in openapi descriptions of `PodIP.IP`  and `HostIP.IP`
    fields to mark the fields used as keys in those lists as required.
```

File: releases/release-1.31/release-notes/maps/pr-126046-map.yaml (hunk 14 above)
Comment (shecodesmagic) [reply releases/release-1.31/release-notes/maps/pr-126046-map.yaml:3]: Used?

File: releases/release-1.31/release-notes/maps/pr-124994-map.yaml [hunk 17]
```diff
@@ -0,0 +1,4 @@
+pr: 124994
+releasenote:
+  text: Added namespace autocompletion for kubectl config set-context command.
```
Comment (cloudmelon) [inline releases/release-1.31/release-notes/maps/pr-124994-map.yaml:3]:
```suggestion
  text: Added namespace autocompletion for `kubectl config set-context` command.
```

File: releases/release-1.31/release-notes/maps/pr-123638-map.yaml (hunk 6 above)
Comment (shecodesmagic) [reply releases/release-1.31/release-notes/maps/pr-123638-map.yaml:3]: Sorry this is wrong, no need to update here.

Comment (cloudmelon) [reply releases/release-1.31/release-notes/maps/pr-123638-map.yaml:3]: ok, thanks

File: releases/release-1.31/release-notes/release-notes-draft.json [hunk 18]
```diff
@@ -229,8 +220,29 @@
       "instrumentation"
     ],
     "duplicate": true,
-    "duplicate_kind": true,
-    "is_mapped": true
+    "duplicate_kind": true
+  },
+  "122025": {
```
Comment (npolshakova) [inline releases/release-1.31/release-notes/release-notes-draft.json:225]: Example of new release note that linked to the map.

File: releases/release-1.31/release-notes/release-notes-draft.json [hunk 19]
```diff
@@ -250,13 +262,12 @@
     "sigs": [
       "cli"
     ],
-    "duplicate_kind": true,
-    "is_mapped": true
```
Comment (npolshakova) [inline releases/release-1.31/release-notes/release-notes-draft.json:254]: Example of `is_mapped` value.

File: releases/release-1.31/release-notes/release-notes-draft.json [hunk 20]
```diff
@@ -34,13 +33,12 @@
       "cli",
       "windows"
     ],
-    "duplicate": true,
-    "is_mapped": true
```
Comment (npolshakova) [inline releases/release-1.31/release-notes/release-notes-draft.json:38]: This was remove, you want to go back in this diff commit and add the `is_mapped": true` values back.

File: releases/release-1.31/release-notes/maps/pr-122025-map.yaml [hunk 21]
```diff
@@ -0,0 +1,4 @@
+pr: 122025
+releasenote:
+  text: `lastSuccessfullTime` in cronjobs will now be set reliably.
```
Comment (npolshakova) [inline releases/release-1.31/release-notes/maps/pr-122025-map.yaml:3]: Having an initial backtick invalid yaml and will break: https://github.com/kubernetes/release/issues/2753

```suggestion
  text: "`lastSuccessfullTime` in cronjobs will now be set reliably."
```

File: releases/release-1.31/release-notes/maps/pr-123587-map.yaml (hunk 5 above)
Comment (npolshakova) [reply releases/release-1.31/release-notes/maps/pr-123587-map.yaml:3]: It's what the author originally had: https://github.com/kubernetes/kubernetes/pull/123587

File: releases/release-1.31/release-notes/maps/pr-126012-map.yaml [hunk 22]
```diff
@@ -0,0 +1,4 @@
+pr: 126012
+releasenote:
+  text: Stoped using wmic on Windows to get uuid in the kubelet.
```
Comment (npolshakova) [inline releases/release-1.31/release-notes/maps/pr-126012-map.yaml:3]:
```suggestion
  text: Stopped using wmic on Windows to get uuid in the kubelet.
```

**Review by npolshakova — COMMENTED** [review]
Looks good! Just a couple last comments to resolve and this should be good to go!

File: releases/release-1.31/release-notes/release-notes-draft.json (hunk 18 above)
Comment (cloudmelon) [reply releases/release-1.31/release-notes/release-notes-draft.json:234]: thanks !

**Comment by cloudmelon** [comment]
> Looks good! Just a couple last comments to resolve and this should be good to go!

great news ! Let me doublecheck everything and squash them into 1 big fat commit !

**Review by npolshakova — APPROVED** [review]
Looks great! Thanks for taking care of all the edits and squashing them into one commit @cloudmelon 🎉

**Review by saschagrunert — APPROVED** [review]

**Pull request merged**